packages/react-storage/src/components/StorageBrowser/Controller.tsx

```diff
@@ -13,5 +14,29 @@ export function Controller(): null {
     handleListLocations({ options: { pageSize: 1000, refresh: true } });
   }, [handleListLocations]);
 
+  const [{ selected }] = useControl({ type: 'ACTION_SELECT' });
+  const [{ history, path }] = useControl({
+    type: 'NAVIGATE',
+  });
+  const [, handleListLocationItems] = useAction({
+    type: 'LIST_LOCATION_ITEMS',
+  });
+
+  const currentPosition = history.length;
+  const hasHistory = !!currentPosition;
+
+  React.useEffect(() => {
+    if (!hasHistory) return;
+
+    // Check to make sure that we're not in an action view
+    // This makes it so that the list locations refreshes when we go back from an action view
+    if (!selected.type) {
```

Comment:
Added this because I needed a way to refresh the location detail view table when coming back from the upload actions view. Is there a better way to do this?

Reply:
Is this issue present previous to this PR?